### PR TITLE
[#1016] Create topics with k8s job

### PR DESCRIPTION
**Pull request by ljupcovangelski**
A Kubernetes job is provisioned with helm. The job has an init container which waits for the kafka service to be there, then executes the main job to create the topics.

### CODE REVIEW

File: infrastructure/helm-chart/charts/provisioning/templates/kafka.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    kafka-topics.sh --create --if-not-exists --zookeeper "${ZOOKEEPER}" --replication-factor "${REPLICAS}" --partitions "${PARTITIONS}" --topic "${AIRY_CORE_NAMESPACE}application.communication.webhooks"
+
+    kafka-topics.sh --create --if-not-exists --zookeeper "${ZOOKEEPER}" --replication-factor "${REPLICAS}" --partitions "${PARTITIONS}" --topic "${AIRY_CORE_NAMESPACE}ops.application.health" --config retention.ms=3600000
+
+    kafka-topics.sh --create --if-not-exists --zookeeper "${ZOOKEEPER}" --replication-factor "${REPLICAS}" --partitions "${PARTITIONS}" --topic "${AIRY_CORE_NAMESPACE}source.facebook.events"
+
+    kafka-topics.sh --create --if-not-exists --zookeeper "${ZOOKEEPER}" --replication-factor "${REPLICAS}" --partitions "${PARTITIONS}" --topic "${AIRY_CORE_NAMESPACE}source.google.events"
+
+    kafka-topics.sh --create --if-not-exists --zookeeper "${ZOOKEEPER}" --replication-factor "${REPLICAS}" --partitions "${PARTITIONS}" --topic "${AIRY_CORE_NAMESPACE}source.twilio.events"
+---
```
Comment (ljupcovangelski) [inline infrastructure/helm-chart/charts/provisioning/templates/kafka.yaml:51]: The script is in a configmap, we need to find a way to generate it there or we go for a different solution.

File: infrastructure/helm-chart/charts/provisioning/templates/kafka.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+          configMap:
+            name: provisioning-scripts
+        - name: kafka-scripts
+          configMap:
+            name: kafka-scripts
+        - name: kafka-config
+          configMap:
+            name: kafka-config
+      restartPolicy: Never
+  backoffLimit: 3
```
Comment (ljupcovangelski) [inline infrastructure/helm-chart/charts/provisioning/templates/kafka.yaml:90]: Waits for kafka to be there, then provisions the topics.

File: infrastructure/helm-chart/charts/provisioning/templates/scripts.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  name: provisioning-scripts
+data:
+  wait-for-service.sh: |
+    #!/bin/sh
+    service=${1}
+    port=${2}
+    delay=${3}
+    name=${4}
+
+    while ! nc -z ${service} ${port} >/dev/null 2>&1; do sleep ${delay}; echo Waiting for ${name} to start...; done
```
Comment (ljupcovangelski) [inline infrastructure/helm-chart/charts/provisioning/templates/scripts.yaml:13]: A general helper function, for waiting for services, to be used in `init containers`.

File: infrastructure/scripts/status.sh [hunk 4]
```diff
@@ -22,7 +22,7 @@ NGROK_ENABLED=$(kubectl get configmap core-config -o jsonpath='{.data.NGROK_ENAB
 if [ "${NGROK_ENABLED}" = "true" ]; then
     FACEBOOK_WEBHOOK_PUBLIC_URL="https://fb-${CORE_ID}.tunnel.airy.co/facebook"
     GOOGLE_WEBHOOK_PUBLIC_URL="https://gl-${CORE_ID}.tunnel.airy.co/google"
-    TWILIO_WEBHOOK_PUBLIC_URL="https://tw-${CORE_ID}.tunnel.airy.co"
+    TWILIO_WEBHOOK_PUBLIC_URL="https://tw-${CORE_ID}.tunnel.airy.co/twilio"
```
Comment (ljupcovangelski) [inline infrastructure/scripts/status.sh:25]: We had a bug there.

**Review by pascal-airy — DISMISSED** [review]
:shipit:

**Review by pascal-airy — APPROVED** [review]

**Pull request merged**